### PR TITLE
fix: Fix subscriptions sending twice

### PR DESCRIPTION
**Pull request by webjunkie**
## Problem

Daily subscriptions are sent twice.

Theory: Since they run 5 minutes before the full hour, next delivery date will be scheduled again on same day (one hour later). Only then the `now` is larger then the intended date and next run will be scheduled correctly in a day.

Follow up from #21419

## Changes

Revert  #21419 but add `max()` so that we never take a date in the past.

## Does this work well for both Cloud and self-hosted?

it doesn't have an impact

## How did you test this code?

- Cover both this case and the original case in tests.

### CODE REVIEW

File: posthog/models/subscription.py [hunk 1]
```diff
@@ -130,6 +130,8 @@ def save(self, *args, **kwargs) -> None:
         # Only if the schedule has changed do we update the next delivery date
         if not self.id or str(self._rrule) != str(self.rrule):
             self.set_next_delivery_date()
+            if "update_fields" in kwargs:
+                kwargs["update_fields"].append("next_delivery_date")
```
Comment (webjunkie) [inline posthog/models/subscription.py:134]: This is now recommended to add in Django 4.2, so doing it here while on it.

https://docs.djangoproject.com/en/4.2/releases/4.2/#setting-update-fields-in-model-save-may-now-be-required

**Review by mariusandra — APPROVED** [review]

**Comment by mariusandra** [comment]
Seems like there's a legit test failure:

image

**Comment by webjunkie** [comment]
Yes, saw it after all 😅 
Buried after some other stuff... which I thought is CI trouble.

**Comment by neilkakkar** [comment]
yeah a fix for this CI spew is here: https://github.com/PostHog/posthog/pull/21613 - getting there

**Pull request merged**